### PR TITLE
Support IPv6 addresses properly

### PR DESCRIPTION
**Pull request by markozajc**
support the `[ipv6]:port` and `ipv6` formats
silly cat forgot that ipv6 is a thing

### CODE REVIEW

**Comment by Anuken** [comment]
> silly cat forgot that ipv6 is a thing

My network doesn't support IPv6, and so few people use it that I just didn't care.

File: core/src/mindustry/ui/dialogs/JoinDialog.java [hunk 1]
```diff
@@ -563,25 +563,32 @@ private void saveServers(){
         transient Host lastHost;
 
         void setIP(String ip){
-
-            //parse ip:port, if unsuccessful, use default values
-            if(ip.lastIndexOf(':') != -1 && ip.lastIndexOf(':') != ip.length() - 1){
-                try{
+            try{
+                boolean isIpv6 = ip.chars().filter(c -> c == ':').count() > 1;
```
Comment (Anuken) [inline core/src/mindustry/ui/dialogs/JoinDialog.java:567]: `chars()` is a Java 8 feature. Don't use it.

**Comment by markozajc** [comment]
tunnelbroker.net :)

File: core/src/mindustry/ui/dialogs/JoinDialog.java (hunk 1 above)
Comment (markozajc) [reply core/src/mindustry/ui/dialogs/JoinDialog.java:567]: shit right sorry

Comment (Anuken) [reply core/src/mindustry/ui/dialogs/JoinDialog.java:567]: Note: `Strings#count(String, char)`

Comment (markozajc) [reply core/src/mindustry/ui/dialogs/JoinDialog.java]: done

**Comment by markozajc** [comment]
found a bug h

**Comment by markozajc** [comment]
nevermind it was just the wind

**Pull request merged**